frontend/src/components/navTabs.tsx

```diff
@@ -25,33 +26,73 @@ const NavTabs = (props: Props) => {
   const navigate = useNavigate();
   const activeTab = tabs.find((tab) => location.pathname.includes(tab.path));
   const activeTabPath = activeTab?.path || "";
+
+  // State to manage the visibility of the mobile menu
+  const [mobileMenuVisible, setMobileMenuVisible] = useState(false);
```

Comment:
As we are using mantine ui library, I would suggest you to use "useDisclosure" hook to manage the state for this.
I have already used it in "frontend/src/layouts/components/settingsLayout.tsx" in this file.